src/main/java/com/intel/genomicsdb/GenomicsDBImporter.java

```diff
@@ -23,1615 +23,1665 @@
-    for (Map.Entry<String, FeatureReader<VariantContext>> mapObject :
-        sampleToReaderMap.entrySet()) {
+    /**
+     * Obtain the chromosome intervals for the column partition specified in the loader JSON file
+     * identified by the rank. The information is returned as a string in JSON format
+     * {
+     * "contigs": [
+     * { "chr1": [ 100, 200] },
+     * { "chr2": [ 500, 600] }
+     * ]
+     * }
+     *
+     * @param loaderJSONFile path to loader JSON file
+     * @param rank           rank/partition index
+     * @return chromosome intervals for the queried column partition in JSON format
+     */
+    private static native String jniGetChromosomeIntervalsForColumnPartition(
+            final String loaderJSONFile,
+            final int rank);
+
+    /**
+     * Create TileDB workspace
+     *
+     * @param workspace path to workspace directory
+     * @return status 0 = workspace created,
+     * -1 = path was not a directory,
+     * -2 = failed to create workspace,
+     * 1 = existing directory, nothing changed
+     */
+    private static native int jniCreateTileDBWorkspace(final String workspace);
+
+    /**
+     * Consolidate TileDB array
+     *
+     * @param workspace path to workspace directory
+     * @param arrayName array name
+     */
+    private static native void jniConsolidateTileDBArray(final String workspace, final String arrayName);
+
+    private static VCFHeaderToFile resolveVCFHeaderToFile(final BaseImportConfig baseImportConfig) {
+        List<VCFHeader> headers = new ArrayList<>();
+        ArrayList<String> sampleNames = new ArrayList<>();
+        Map<String, String> sampleNameToFileName = new LinkedHashMap<>();
+
+        //Get merged header first
+        for (String file : baseImportConfig.getFiles()) {
+            AbstractFeatureReader<VariantContext, LineIterator> reader =
+                    AbstractFeatureReader.getFeatureReader(file, new VCFCodec(), false);
+            headers.add((VCFHeader) reader.getHeader());
+            final String sampleName = ((VCFHeader) reader.getHeader()).getGenotypeSamples().get(0);
+            sampleNames.add(sampleName);
+            sampleNameToFileName.put(sampleName, file);
+            //Hopefully, GC kicks in and frees resources assigned to reader
+        }
 
-      String sampleName = mapObject.getKey();
-      FeatureReader<VariantContext> featureReader = mapObject.getValue();
+        return new VCFHeaderToFile(headers, sampleNames, sampleNameToFileName);
+    }
 
-      if (featureReader == null) {
-        throw new IllegalArgumentException("Null FeatureReader found for sample: " + sampleName);
-      }
+    /**
+     * Import multiple chromosome interval
+     * @param baseImportConfig
+     */
+    public static void parallelImport(final BaseImportConfig baseImportConfig) throws IOException {
+        VCFHeaderToFile vcfHeaderToFile = resolveVCFHeaderToFile(baseImportConfig);
```

Comment:
I suspect this function will have to be moved outside the parallelImport function, let me think about this some more.

Reply:
We will have to intercept GATK-4 code here: https://github.com/broadinstitute/gatk/blob/a0efe633782b9f219d5cb1293386ff88fc2f2df5/src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java#L487

I can go over the requirement in GATK-4 if you want.

Reply:
I can do it but before of that, I would like to stress import and FeatureReader functionality.